src/main/java/io/temporal/samples/hello/HelloDetachedCancellationScope.java

```diff
@@ -0,0 +1,226 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.hello;
+
+import io.temporal.activity.Activity;
+import io.temporal.activity.ActivityCancellationType;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowFailedException;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.client.WorkflowStub;
+import io.temporal.failure.ActivityFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Sample Temporal workflow that shows how to run some cleanup code when a workflow has been
+ * explicitly cancelled.
+ *
+ * <p>To execute this example a locally running Temporal service instance is required. You can
+ * follow instructions on how to set up your Temporal service here:
+ * https://github.com/temporalio/temporal/blob/master/README.md#download-and-start-temporal-server-locally
+ */
+public class HelloDetachedCancellationScope {
+
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloDetachedCancellationTaskQueue";
+
+  // Define our workflow unique id
+  static final String WORKFLOW_ID = "HelloDetachedCancellationWorkflow";
+
+  /**
+   * Define the Workflow Interface. It must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow code includes core processing logic. It that shouldn't contain any heavyweight
+   * computations, non-deterministic code, network calls, database operations, etc. All those things
+   * should be handled by Activities.
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface GreetingWorkflow {
+
+    /**
+     * This method is executed when the workflow is started. The workflow completes when the
+     * workflow method finishes execution.
+     */
+    @WorkflowMethod
+    String getGreeting(String name);
+
+    /** Query method to get the greeting */
+    @QueryMethod
+    String queryGreeting();
+  }
+
+  /**
+   * Define the Activity Interface. Activities are building blocks of any temporal workflow and
+   * contain any business logic that could perform long running computation, network calls, etc.
```

Comment:
```suggestion
   * This is the Activity Definition's Interface. Activities are building blocks of any Temporal Workflow and
   * contain any business logic that could perform long running computation, network calls, etc.
```

Reply:
Same here: global change will apply to all hello samples